docs/sourcemaps.asciidoc

```diff
@@ -1,38 +1,192 @@
 
-For a number of reasons, it is common practice to minify JavaScript code, for example to reduce network latency.
-While improving performance, minified code can be hard to debug.
-A source map library helps by mapping the minified files back to the the original source code.
+++++
+<titleabbrev>Create and upload source maps (RUM)</titleabbrev>
+++++
 
-APM Server provides a <<sourcemap-api,Source Map API>> for uploading source maps.
+Minifying JavaScript bundles in production is a common practice;
+it can greatly improve the load time and network latency of your applications.
+The problem with minifying code is that it can be hard to debug.
+Here's an example of an exception stack trace in the APM app when using minified code.
+As you can see, it's not very helpful.
 
-By default, <<config-sourcemapping-enabled>> is set to `true`.
-This means if a source map has previously been uploaded,
-source mapping will automatically be applied all incoming transactions and errors.
+[role="screenshot"]
+image::images/source-map-before.png[APM app without source mapping]
 
+That’s why Elastic APM supports source maps.
+A source map maps minified files back to the original source code, allowing you to maintain the speed advantage of minified code,
+without losing the ability to quickly and easily debug your application.
+Here's the same example as before, but with a source map uploaded and applied:
+
+[role="screenshot"]
+image::images/source-map-after.png[APM app with source mapping]
+
+Follow the steps below to enable source mapping your error stack traces in the APM app:
+
+* <<sourcemap-rum-initialize>>
+* <<sourcemap-rum-generate>>
+* <<sourcemap-rum-upload>>
+
+[float]
+[[sourcemap-rum-initialize]]
+=== Initialize the RUM Agent
+
+First, set the service name and version of your application when initializing the RUM Agent.
+
+[source,js]
+----
+import { init as initApm } from '@elastic/apm-rum'
+
+const apm = initApm({
+  serviceName: 'myService',
+  serviceVersion: serviceVersion
+})
+----
+
+This `serviceVersion` can also be the version from your `package.json`. For example:
+
+[source,js]
+----
+const serviceVersion = require("./package.json").version
+----
+
+Or, it could be a git commit reference. For example:
+
+[source,js]
+----
+const git = require('git-rev-sync')
+const serviceVersion = git.short()
+----
+
+It can also be any other unique string that indicates a specific version of your application.
+APM Server uses the `serviceVersion` to match the correct source map file to each stack trace.
+
+[float]
+[role="child_attributes"]
+[[sourcemap-rum-generate]]
+=== Generate a source map
+
+Source maps can be generated and configured in many different ways.
+To be compatible with Elastic APM, they must follow the
+https://docs.google.com/document/d/1U1RGAehQwRypUTovF1KRlpiOFze0b-_2gc6fAH0KY0k[source map revision 3 proposal spec].
+
+One way to generate and configure sourcemaps, is with webpack.
```

Comment:
This reads a bit like if you want to generate sourcemaps, you can do it with webpack. The reality is more the other way around, like: if you are using webpack, you can generate sourcemaps this way: [....], if you are  using parcel, they are generated already by default, etc.